types/v56/types.go

```diff
@@ -1445,43 +1445,64 @@ type Value struct {
 	Value string `xml:"http://schemas.dmtf.org/ovf/envelope/1 value,attr,omitempty"`
 }
 
+// MetadataValue is the type returned when querying a unique entry of metadata.
+// Type: MetadataValueType
+// Namespace: http://www.vmware.com/vcloud/v1.5
 type MetadataValue struct {
-	XMLName    xml.Name    `xml:"MetadataValue"`
-	Xsi        string      `xml:"xmlns:xsi,attr"`
-	Xmlns      string      `xml:"xmlns,attr"`
-	TypedValue *TypedValue `xml:"TypedValue"`
+	XMLName    xml.Name            `xml:"MetadataValue"`
+	Xsi        string              `xml:"xmlns:xsi,attr"`
+	Xmlns      string              `xml:"xmlns,attr"`
+	Domain     *MetadataDomainTag  `xml:"Domain,omitempty"`
+	TypedValue *MetadataTypedValue `xml:"TypedValue"`
 }
 
-type TypedValue struct {
-	XsiType string `xml:"xsi:type,attr"`
+// MetadataTypedValue is the content of a metadata entry.
+// Type: MetadataTypedValue
+// Namespace: http://www.vmware.com/vcloud/v1.5
+// Description: One of: MetadataStringValue, MetadataNumberValue, MetadataBooleanValue, MetadataDateTimeValue
+// Since: 5.1
+type MetadataTypedValue struct {
```

Comment:
Just to avoid breaking things in non major version you could add a type alias and note it is deprecated in favor of
MetadataTypedValue

`type TypedValue = MetadataTypedValue`

Reply:
You're very right, this could break things. Changed!